src/engine/interop/specification.md

```diff
@@ -0,0 +1,146 @@
+| - | - | - |
+| 4 | Missing resource | Should be used when one of the call paramters references a missing resource and the call can't be processed, e.g. some action over unknown block is requested |
+
+## Structures
+
+*Note:* Fields having `QUANTITY` type are serialized in big-endian.
+
+### ExecutionPayload
+
+This structure maps on the [`ExecutionPayload`](https://github.com/ethereum/consensus-specs/blob/dev/specs/merge/beacon-chain.md#ExecutionPayload) structure of the beacon chain spec. The fields are encoded as follows:
+- `parentHash`: `DATA`, 32 Bytes
+- `coinbase`:  `DATA`, 20 Bytes
+- `stateRoot`: `DATA`, 32 Bytes
+- `receiptRoot`: `DATA`, 32 Bytes
+- `logsBloom`: `DATA`, 256 Bytes
+- `random`: `DATA`, 32 Bytes
+- `blockNumber`: `QUANTITY`, 64 Bits
+- `gasLimit`: `QUANTITY`, 64 Bits
+- `gasUsed`: `QUANTITY`, 64 Bits
+- `timestamp`: `QUANTITY`, 64 Bits
+- `extraData`: `DATA`
+- `baseFeePerGas`: `QUANTITY`, 256 Bits
+- `blockHash`: `DATA`, 32 Bytes
+- `transactions`: `Array of DATA` - Array of transaction objects, each object is a byte list (`DATA`) representing `TransactionType || TransactionPayload` or `LegacyTransaction` as defined in [EIP-2718](https://eips.ethereum.org/EIPS/eip-2718)
+
+## Core
+
+### engine_preparePayload
+
+#### Parameters
+1. `Object` - The payload attributes:
+- `parentHash`: `DATA`, 32 Bytes - hash of the parent block
+- `timestamp`: `QUANTITY`, 64 Bits - value for the `timestamp` field of the new payload
+- `random`: `DATA`, 32 Bytes - value for the `random` field of the new payload
+- `feeRecipient`: `DATA`, 20 Bytes - suggested value for the `coinbase` field of the new payload
+
+#### Returns
+1. `payloadId`: `QUANTITY`, 64 Bits - identifier of the payload building process or an error
+
+#### Specification
+
+1. Given provided field value set client software **SHOULD** build the initial version of the payload which has an empty transaction set.
+
+2. Client software **SHOULD** start the process of updating the payload. The strategy of this process is implementation dependent. The default strategy would be to keep the transaction set up-to-date with the state of local mempool.
+
+3. Client software **SHOULD** stop the updating process either by finishing to serve the [**`engine_getPayload`**](#engine_getPayload) call with the same `payloadId` value or when [`SECONDS_PER_SLOT`](https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/beacon-chain.md#time-parameters-1) (currently set to 12 in the Mainnet configuration) seconds have passed since the point in time identified by the `timestamp` parameter.
+
+4. Client software **MUST** set the payload field values according to the set of parameters passed in the call to this method with exception for the `feeRecipient`. The `coinbase` field value **MAY** stem from what is specified by the `feeRecipient` parameter.
+
+5. Client software **SHOULD** respond with `2: Action not allowed` error if the sync process is in progress.
```

Comment:
Why SHOULD here?

Reply:
I think because client developers could decide to reach out to a external source for an execution payload if one is requested while syncing? Is that a behavior that we want to explicitly forbid?

Reply:
Yeah, it could either request a payload from builders or build an empty payload with the parent's state root